pytensor/link/pytorch/dispatch/subtensor.py

```diff
@@ -0,0 +1,38 @@
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.subtensor import (
+    AdvancedSubtensor,
+    AdvancedSubtensor1,
+    Subtensor,
+    indices_from_subtensor,
+)
+
+
+@pytorch_funcify.register(Subtensor)
+@pytorch_funcify.register(AdvancedSubtensor)
+@pytorch_funcify.register(AdvancedSubtensor1)
+def pytorch_funcify_Subtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    def subtensor(x, *ilists):
+        indices = indices_from_subtensor(ilists, idx_list)
+        new_indices = []
+
+        for i in indices:
+            new_start, new_step, new_stop = None, None, None
+            if isinstance(i, slice):
+                if i.start:
+                    new_start = i.start.item()
+                if i.step:
+                    new_step = i.step.item()
+                if i.stop:
+                    new_stop = i.stop.item()
+                new_indices.append(slice(new_start, new_stop, new_step))
+            else:
+                new_indices.append(i.tolist())
```

Comment:
Why is `.tolist()` needed, and what happens if there's a 2d index?

Reply:
Have not tried with a 2d index yet, but `.tolist` is needed for such cases:
`pt_subtensor.advanced_subtensor1(x_pt, [1, 2])`

Reply:
Torch doesn't accept tensors as indices? Does it require converting them to a list? Otherwise we shouldn't have to to be doing this conversion

Reply:
There is a shape mismatch when passing tensors as indices

Reply:
Can you provide more details? When I run the following code it works fine:

```python
import torch
x = torch.arange(5*3).reshape((5, 3))
x[torch.tensor([0, 2, 4])]
```
I did not need to convert the tensor indices with .tolist(), which should be the same case here?

Reply:
In this case the indices are torch tensors, we have a list of tensors instead of a one dimensional tensor

Reproducible code:
```
import torch
x = torch.arange(5*3*5*4).reshape((5, 3, 5, 4))
a = x[torch.tensor([0, 2, 4])]

indices = (torch.tensor(0), torch.tensor(2), torch.tensor(4))

b = x[indices]

print(a.shape) # torch.Size([3, 3, 5, 4])
print(b.shape) # torch.Size([4])
```

Reply:
Sure but that also doesn't require any call to .tolist()

Reply:
We could use `.item` instead of `.tolist` but that would not work cases like: `pt.subtensor.advanced_subtensor1(x, [1, 2])`

Reply:
I think we need to take a step back here. This should be simpler. 

First let's separate Subtensor from AdvancedSubtensor.

Subtensor only allows basic indexing and no array indices. This shouldn't need any type conversion.

Then AdvancedSubtensor allows both, but distinctively you don't need to create slice objects as those are inputs to the Op, unlike the basic Subtensor where you have to create them as the inputs are just start, stop and step of the slices.